daemon/cmd/daemon_main.go

```diff
@@ -1737,6 +1739,46 @@ func newDaemonPromise(params daemonParams) (promise.Promise[*Daemon], promise.Pr
 	return daemonPromise, cfgPromise
 }
 
+type syncCRDsPromiseParams struct {
+	cell.In
+
+	Lifecycle cell.Lifecycle
+	Clientset k8sClient.Clientset
+	Resources *k8sSynced.Resources
+	APIGroups *k8sSynced.APIGroups
+}
+
+func newSyncCRDsPromise(params syncCRDsPromiseParams) promise.Promise[resource.CRDSync] {
+	crdSyncResolver, crdSyncPromise := promise.New[resource.CRDSync]()
+	if !params.Clientset.IsEnabled() || option.Config.DryMode {
+		crdSyncResolver.Reject(fmt.Errorf("disabled"))
```

Comment:
Can we provide a more detailed error message, as this would likely end up being logged to the user.

Reply:
Now have it like:
```
	crdSyncResolver.Reject(fmt.Errorf("CRDSync promise is disabled in tests"))
```